twine/utils.py

```diff
@@ -195,7 +195,7 @@ def get_userpass_value(cli_value, config, key, prompt_strategy=None):
     """
     if cli_value is not None:
         return cli_value
-    elif config.get(key):
+    elif key in config:
```

Comment:
Given that config always defaults with `{'username': None, 'password': None}`, I now believe this change unconditionally disables prompts.

Reply:
If you don't add `username` nor `password` keys in your configuration file, the keys won't be available in the parsed config object, so it should continue to the prompt_strategy part.

Reply:
That's [where you're mistaken](https://github.com/pypa/twine/blob/61d01ac125c047ae6d6e5e4aab6ffa96cd3a6b95/twine/utils.py#L67).

Reply:
What if I don't have a `server-login` section (which is my case):
```
[distutils]
index-servers =
    internal

[internal]
repository: ******
username
password
```

Reply:
The defaults [still get copied into every repository](https://github.com/pypa/twine/blob/61d01ac125c047ae6d6e5e4aab6ffa96cd3a6b95/twine/utils.py#L85).